docs/howto/stack_version_support.md

```diff
@@ -0,0 +1,80 @@
+Intro...
+
+TBD: When to reconsider the version constraint.
+
+
+Support for all versions since the release of the package
+----
+Example: `>=7.16.0`.
+
+With this approach, the released package can be used in any version since the
+release of the package.
+
+Pros:
+* No need to keep multiple development branches, no need for backports.
+
+Cons:
+* False premise, difficult to ensure compatibility with future majors.
+
+Recommendation: Not recommended.
+
+Support for multiple majors
+----
+Example: `^7.14.0 || ^8.0.0`.
+
+With this approach, the released package can be used in any tested major version
+since the release of the package.
+
+Pros:
+* No need to keep multiple development branches, no need for backports, same
+  package for 7.x and 8.x.
+* Less disruptive for users.
+
+Cons:
+* Testing should be done with more versions.
```

Comment:
nit: manual testing

Reply:
I refer to testing in general in all appearances of the term here. In general we should be able to run CI with multiple versions, independently of them being 7.16.0 and 8.1.0, or 8.0.0 and 8.2.0.
The difference here is that the same package is supporting more versions than the alternative where there would be a package for 7.x versions and another one for 8.x. Though at the end the difference may be not so big.

Reply:
I think you can add an issue to the backlog to implement the support matrix which will cover next versions. I suppose it should be executing in the background, not blocking any code pushes, just to prevent few hours CI runs.

Reply:
Yeah, good point, added: https://github.com/elastic/integrations/issues/2134